lib/primitiveToGeometry.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+var readAccessor = require('./readAccessor');
+var numberOfComponentsForType = require('./numberOfComponentsForType');
+var modeToPrimitiveType = require('./modeToPrimitiveType');
+var Cesium = require('cesium');
+var Geometry = Cesium.Geometry;
+var GeometryAttribute = Cesium.GeometryAttribute;
+var ComponentDatatype = Cesium.ComponentDatatype;
+var Cartesian3 = Cesium.Cartesian3;
+
+module.exports = primitiveToGeometry;
+
+function primitiveToGeometry(gltf, primitive) {
+    var positionId = primitive.attributes.POSITION;
+    var positionAccessor = gltf.accessors[positionId];
+    var componentsPerAttribute = numberOfComponentsForType(positionAccessor.type);
+    var values = readAccessor(gltf, positionAccessor);
+    
+    var packedValues = [];
+    packedValues.length = values.data.length * 3;
```

Comment:
Change this to `var packedValues = new Array(values.data.length * 3);`
You could also factor out `values.data.length` into its own variable and use it here and in the for loop below.